workspaces/arborist/lib/yarn-lock.js

```diff
@@ -33,10 +32,45 @@ const consistentResolve = require('./consistent-resolve.js')
 const { dirname } = require('path')
 const { breadth } = require('treeverse')
 
+// Sort Yarn entries respecting the yarn.lock sort order
+const yarnEntryPriorities = {
+  name: 1,
+  version: 2,
+  uid: 3,
+  resolved: 4,
+  integrity: 5,
+  registry: 6,
+  dependencies: 7,
+}
+
+const priorityThenLocaleCompare = (a, b) => {
+  if (yarnEntryPriorities[a] || yarnEntryPriorities[b]) {
+    return (yarnEntryPriorities[a] || 100) > (yarnEntryPriorities[b] || 100) ? 1 : -1
+  }
+
+  return localeCompare(a, b)
+}
+
+const shouldQuoteString = str => {
+  return str.indexOf('true') === 0 ||
```

Comment:
is there a reason this isn't a test of equality with the string `true` or `false`?